From 12ef4a1c9a0019626b6c3341f6c197b5910dc471 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:00 +0800
Subject: [PATCH 007/150] arm64: kpti: Add Phytium FTC3xx/6xx CPU cores to kpti
 safelist

Since Phytium FTC3xx/6xx CPU cores are all meltdown safe, add them to
kpti_safe_list[].

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I8db79ef0b2513c8dfd9c5f7cbf03ceabb829da16
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 arch/arm64/kernel/cpufeature.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/kernel/cpufeature.c b/arch/arm64/kernel/cpufeature.c
index 7e9660455900..cc55214d4d3a 100644
--- a/arch/arm64/kernel/cpufeature.c
+++ b/arch/arm64/kernel/cpufeature.c
@@ -1686,6 +1686,12 @@ static bool unmap_kernel_at_el0(const struct arm64_cpu_capabilities *entry,
 		MIDR_ALL_VERSIONS(MIDR_CORTEX_A73),
 		MIDR_ALL_VERSIONS(MIDR_HISI_TSV110),
 		MIDR_ALL_VERSIONS(MIDR_NVIDIA_CARMEL),
+		MIDR_ALL_VERSIONS(MIDR_PHYTIUM_FTC310),
+		MIDR_ALL_VERSIONS(MIDR_PHYTIUM_FTC660),
+		MIDR_ALL_VERSIONS(MIDR_PHYTIUM_FTC661),
+		MIDR_ALL_VERSIONS(MIDR_PHYTIUM_PS17064),
+		MIDR_ALL_VERSIONS(MIDR_PHYTIUM_FTC663),
+		MIDR_ALL_VERSIONS(MIDR_PHYTIUM_FTC664),
 		MIDR_ALL_VERSIONS(MIDR_QCOM_KRYO_2XX_GOLD),
 		MIDR_ALL_VERSIONS(MIDR_QCOM_KRYO_2XX_SILVER),
 		MIDR_ALL_VERSIONS(MIDR_QCOM_KRYO_3XX_SILVER),
-- 
2.47.0

